spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -744,12 +786,6 @@ def create_shortcuts(self):
             name='Clear shell',
             parent=self)
 
-        restart_kernel = self.config_shortcut(
-            self.ipyclient.restart_kernel,
-            context='ipython_console',
-            name='Restart kernel',
-            parent=self)
```

Comment:
This shortcut is important, so please find a way to restore it. I know many users (and also I) use it all the time to quickly restart the kernel.

Reply:
but it still work, doesn't it? The same shortcut is created in main_widget.py:
```
self.restart_action = self.create_action(
            IPythonConsoleWidgetActions.Restart,
            text=_("Restart kernel"),
            icon=self.create_icon('restart'),
            triggered=self.restart_kernel,
            register_shortcut=True
        )
```

Reply:
> but it still work, doesn't it?

Nop, it doesn't. I changed it to other shortcuts and they do nothing when pressed on the console. Probably it's necessary to have it here too so that it actually has effect.

Reply:
That is strange. When launching this PR, I can:

1 - click on the console
2 - "Cmd + ."
3 - A pop up appears asking if I want to restart the kernel
4 - Open settings and change the shortcut to "Cmd + o"
5 - click on the console
6 - "Cmd + o"
7 - A pop up appears asking if I want to restart the kernel

That is not what you are observing? I mean `create_action(register_shortcut=True)` should work and does work in my case, could i be a linux-only bug?

Reply:
> That is not what you are observing?

Nop. After I change the shortcut in Preferences, it doesn't have any effect in the console.

> could it be a linux-only bug?

Possibly. @dalthviz, could you check if you're able to change the "restart kernel" shortcut on Windows?

Reply:
A possible cause for this is that the shortcut is not registered anymore in `ShellWidget`. That's how I understand Qt triggers it when the focus widget is precisely that one and the reason why we had this piece of code you removed here (actually, I don't understand how a different shortcut is working for you on Mac).

Reply:
ctrl + . is hardcoded in qtconsole, so I skipped this. It should work now

Reply:
It still doesn't work for me @impact27.

Reply:
This is perplexing. I will investigate but because on Mac ctrl + . And cmd + . Are distinct, it is not trivial for me to debug this.